applications/ShapeOptimizationApplication/python_scripts/communicator_factory.py

```diff
@@ -25,8 +25,11 @@ def __init__(self, optimization_settings):
         self.supported_constraint_types = ["=", "<", ">", "<=", ">="]
         self.supported_constraint_references = ["initial_value", "specified_value"]
 
-        self.__initializeListOfRequests()
-        self.__initializeListOfResponses()
+        self.list_of_requests = self.__initializeListOfRequestsRecursively(optimization_settings["objectives"])
```

Comment:
also here, there is no need for both these functions to have the recursive logic, as they simply do the same on all the reposnes that are passed. One function to extract a list of all existing reposnses that are then passed to these more simple funcitons would be enough.